File: app/src/main/java/com/stratagile/qlink/ui/activity/main/component/TestComponent.java
package com.stratagile.qlink.ui.activity.main.component;

import com.stratagile.qlink.application.AppComponent;
import com.stratagile.qlink.ui.activity.base.ActivityScope;
import com.stratagile.qlink.ui.activity.main.TestActivity;
import com.stratagile.qlink.ui.activity.main.module.TestModule;

import dagger.Component;

/**
 * @author hzp
 * @Package com.stratagile.qlink.ui.activity.main
 * @Description: The component for TestActivity
 * @date 2018/12/18 11:09:36
 */
@ActivityScope
@Component(dependencies = AppComponent.class, modules = TestModule.class)
public interface TestComponent {
    TestActivity inject(TestActivity Activity);
}

File: app/src/main/java/com/stratagile/qlink/ui/activity/test/component/TestComponent.kt
package com.stratagile.qlink.ui.activity.test.component

import com.stratagile.qlink.application.AppComponent
import com.stratagile.qlink.ui.activity.base.ActivityScope
import com.stratagile.qlink.ui.activity.test.TestFragment
import com.stratagile.qlink.ui.activity.test.module.TestModule

import dagger.Component

/**
 * @author hzp
 * @Package com.stratagile.qlink.ui.activity.test
 * @Description: The component for TestFragment
 * @date 2018/09/10 16:51:54
 */
@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(TestModule::class))
interface TestComponent {
    fun inject(Fragment: TestFragment): TestFragment
}